### PR TITLE
ThreatVault EDL Feed

### PR DESCRIPTION
**Pull request by Ni-Knight**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
related: https://jira-dc.paloaltonetworks.com/browse/CIAC-10961

## Description
This is a feed based on PANW ThreatVault EDLs

### CODE REVIEW

File: Packs/FeedThreatVault/Integrations/FeedThreatVault/FeedThreatVault.py [hunk 1]
```diff
@@ -0,0 +1,353 @@
...
+            demisto.setLastRun({"last_successful_run": run_datetime})
+
+        elif command in commands:
+            return_results(commands[command](client, list_format, demisto.args()))
+        else:
+            raise NotImplementedError(f'Command "{command}" was not implemented.')
+
+    except Exception:
+        demisto.error(traceback.format_exc())  # print the traceback
+        return_error(f"Failed to execute {command} command. The command not implemented")
```
Comment (samuelFain) [inline Packs/FeedThreatVault/Integrations/FeedThreatVault/FeedThreatVault.py:349]: Please note that any exception raised from the code would be excepted here and printed with the "The command not implemented" suffix - which may not always be suitable.

File: Packs/FeedThreatVault/Integrations/FeedThreatVault/FeedThreatVault.py [hunk 2]
```diff
@@ -0,0 +1,353 @@
...
+        return CommandResults(
+            outputs_prefix=f'{INTEGRATION_ENTRY_CONTEXT}.{CONTEXT_KEY}',
+            outputs_key_field='',
+            outputs=table_full,
+            readable_output=markdown,
+            raw_response=response
+        )
+
+    else:
+        raise DemistoException(f"couldn't fetch, {response.get('success')}, response.get('message')")
```
Comment (samuelFain) [inline Packs/FeedThreatVault/Integrations/FeedThreatVault/FeedThreatVault.py:199]: If we got to this section in the code, than response.get('success') is None, so what value does it add to print it?
Maybe we could print the error code or error message instead?

File: Packs/FeedThreatVault/Integrations/FeedThreatVault/FeedThreatVault.py [hunk 3]
```diff
@@ -0,0 +1,353 @@
...
+                listformat=list_format,
+                offset=offset,
+            )
+
+            response = client.get_indicators_request(args=query)
+            ipaddr_list.extend(response.get("data", {}).get("ipaddr"))
+            offset += LIMIT
+
+    else:
+        raise DemistoException(f"couldn't fetch, {response.get('success')}, response.get('message')")
```
Comment (samuelFain) [inline Packs/FeedThreatVault/Integrations/FeedThreatVault/FeedThreatVault.py:270]: Same here

**Review by samuelFain — CHANGES_REQUESTED** [review]
Nice! Please see my comments.

File: Packs/FeedThreatVault/Integrations/FeedThreatVault/FeedThreatVault.yml [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+  - interval
+  - indicatorType
+  - suddenDeath
+  section: Collect
+- display: ''
+  name: feedExpirationInterval
+  defaultvalue: '20160'
+  type: 1
+  required: false
+  section: Collect
```
[the remark this answers is not in the document]
Comment (Ni-Knight) [reply Packs/FeedThreatVault/Integrations/FeedThreatVault/FeedThreatVault.yml:106]: Removed it, shows default
image

File: Packs/FeedThreatVault/Integrations/FeedThreatVault/FeedThreatVault.py (hunk 2 above)
Comment (Ni-Knight) [reply Packs/FeedThreatVault/Integrations/FeedThreatVault/FeedThreatVault.py:199]: Yep, thats a typo, forgot the curly brackets, fixed.

File: Packs/FeedThreatVault/Integrations/FeedThreatVault/FeedThreatVault.py (hunk 3 above)
Comment (Ni-Knight) [reply Packs/FeedThreatVault/Integrations/FeedThreatVault/FeedThreatVault.py:270]: same fix

File: Packs/FeedThreatVault/Integrations/FeedThreatVault/FeedThreatVault.py (hunk 1 above)
Comment (Ni-Knight) [reply Packs/FeedThreatVault/Integrations/FeedThreatVault/FeedThreatVault.py:349]: True, added better handling of it.

File: Packs/FeedThreatVault/Integrations/FeedThreatVault/FeedThreatVault.yml (hunk 4 above)
[the remark this answers is not in the document]
Comment (Ni-Knight) [reply Packs/FeedThreatVault/Integrations/FeedThreatVault/FeedThreatVault.yml]: apparently you have to leave it that way, formatter demanded it.

File: Packs/FeedThreatVault/Integrations/FeedThreatVault/FeedThreatVault.yml [hunk 5]
```diff
@@ -0,0 +1,172 @@
...
+- display: Indicator Reputation
+  name: feedReputation
+  type: 18
+  required: false
+  options:
+  - None
+  - Good
+  - Suspicious
+  - Bad
+  additionalinfo: Indicators from this integration instance will be marked with this reputation
```
Comment (ShirleyDenkberg) [inline Packs/FeedThreatVault/Integrations/FeedThreatVault/FeedThreatVault.yml:74]:
```suggestion
  additionalinfo: Indicators from this integration instance will be marked with this reputation.
```

File: Packs/FeedThreatVault/Integrations/FeedThreatVault/FeedThreatVault.yml [hunk 6]
```diff
@@ -0,0 +1,172 @@
...
+  type: 15
+  required: true
+  options:
+  - A - Completely reliable
+  - B - Usually reliable
+  - C - Fairly reliable
+  - D - Not usually reliable
+  - E - Unreliable
+  - F - Reliability cannot be judged
+  additionalinfo: Reliability of the source providing the intelligence data
```
Comment (ShirleyDenkberg) [inline Packs/FeedThreatVault/Integrations/FeedThreatVault/FeedThreatVault.yml:88]:
```suggestion
  additionalinfo: Reliability of the source providing the intelligence data.
```

File: Packs/FeedThreatVault/Integrations/FeedThreatVault/FeedThreatVault.yml [hunk 7]
```diff
@@ -0,0 +1,172 @@
...
+    - name: version
+      description: The version of the EDL to retrieve.
+      required: true
+      auto: PREDEFINED
+      predefined:
+      - latest
+      defaultValue: latest
+    outputs:
+    - contextPath: ThreatVault.EDL.ipaddr
+      description: EDL Indicators.
```
Comment (ShirleyDenkberg) [inline Packs/FeedThreatVault/Integrations/FeedThreatVault/FeedThreatVault.yml:153]:
```suggestion
      description: EDL indicators.
```

File: Packs/FeedThreatVault/Integrations/FeedThreatVault/FeedThreatVault.yml [hunk 8]
```diff
@@ -0,0 +1,172 @@
...
+      type: String
+    - contextPath: ThreatVault.EDL.name
+      description: EDL Name.
+      type: string
+    - contextPath: ThreatVault.EDL.content_version
+      description: Content Version number.
```
Comment (ShirleyDenkberg) [inline Packs/FeedThreatVault/Integrations/FeedThreatVault/FeedThreatVault.yml:159]:
```suggestion
      description: EDL name.
      type: string
    - contextPath: ThreatVault.EDL.content_version
      description: Content version number.
```

File: Packs/FeedThreatVault/Integrations/FeedThreatVault/FeedThreatVault_description.md [hunk 9]
```diff
@@ -0,0 +1,10 @@
+## Get Your API Key
+  1. Log in to the [Palo Alto Networks Customer Service Portal](https://support.paloaltonetworks.com/Support) with an account that has Super User privileges.
+  2. Select **Assets** and then **API Key Management**.
+  3. From the **Select an API key** drop down, select the API key based on the subscription type.
```
Comment (ShirleyDenkberg) [inline Packs/FeedThreatVault/Integrations/FeedThreatVault/FeedThreatVault_description.md:4]:
```suggestion
  3. From the **Select an API key** dropdown list, select the API key based on the subscription type.
```

File: Packs/FeedThreatVault/Integrations/FeedThreatVault/FeedThreatVault_description.md [hunk 10]
```diff
@@ -0,0 +1,10 @@
...
+  4. Set your API key in the API key field of the integration configuration
+   
+See [this documentation](https://support.paloaltonetworks.com/Support) for information about the API key.
```
Comment (ShirleyDenkberg) [inline Packs/FeedThreatVault/Integrations/FeedThreatVault/FeedThreatVault_description.md:7]: This is not a document.  
How about:
See [here}(https://support.paloaltonetworks.com/Support) for information about the API key.
(However, I am not sure what I should click once I get to this page.)

File: Packs/FeedThreatVault/Integrations/FeedThreatVault/README.md [hunk 11]
```diff
@@ -0,0 +1,31 @@
...
+* **threatvault-get-indicators**: Manually fetch indicators from the PANW Threat Vault feed.
+
+## Troubleshooting
+
+If you encounter any issues:
+1. Verify your API key is correct and has the necessary permissions.
+2. Check the integration's logs for any error messages.
+3. Ensure your network allows outbound connections to the PANW Threat Vault API endpoint.
+
+For more information on using this integration, please refer to the PANW Threat Vault documentation.
```
Comment (ShirleyDenkberg) [inline Packs/FeedThreatVault/Integrations/FeedThreatVault/README.md:31]:
```suggestion
For more information on using this integration, refer to the PANW Threat Vault documentation.
```
Is there a link we can add here?

**Comment by ShirleyDenkberg** [comment]
@samuelFain Doc review completed.

File: Packs/FeedThreatVault/Integrations/FeedThreatVault/README.md (hunk 11 above)
Comment (Ni-Knight) [reply Packs/FeedThreatVault/Integrations/FeedThreatVault/README.md:31]: Not that I know of.

**Review by samuelFain — APPROVED** [review]
Nice!

**Pull request merged**